chainer_chemistry/training/extensions/roc_auc_evaluator.py

```diff
@@ -0,0 +1,128 @@
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import reporter
+from chainer.dataset import convert
+from chainer.training.extensions import Evaluator
+from sklearn import metrics
+
+
+def _get_1d_numpy_array(v):
+    """Convert array or Variable to 1d numpy array
+    
+    Args:
+        v (numpy.ndarray or cupy.ndarray or chainer.Variable): array to be
+            converted to 1d numpy array 
+
+    Returns (numpy.ndarray): Raveled 1d numpy array
+
+    """
+    if isinstance(v, chainer.Variable):
+        v = v.data
+    return cuda.to_cpu(v).ravel()
+
+
+def _to_list(a):
+    """convert value `a` to list
+    
+    Args:
+        a: value to be convert to `list`
+
+    Returns (list):
+
+    """
+    if isinstance(a, (int, float)):
+        return [a, ]
+    else:
+        # expected to be list or some iterable class
+        return a
+
+
+class ROCAUCEvaluator(Evaluator):
+
+    """Evaluator which calculates ROC AUC score
```

Comment:
I would like to add ' this implementation is restricted to the binary classification task'.

Reply:
ok